dev/PushNotifications/PushNotificationChannel.h

```diff
@@ -3,24 +3,39 @@
 
 #pragma once
 #include "Microsoft.Windows.PushNotifications.PushNotificationChannel.g.h"
+#include <NotificationsLongRunningProcess_h.h>
 
 namespace winrt::Microsoft::Windows::PushNotifications::implementation
 {
-    struct PushNotificationChannel : PushNotificationChannelT<PushNotificationChannel>
+    typedef winrt::Windows::Foundation::TypedEventHandler<
+        winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel,
+        winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> PushNotificationEventHandler;
+
+    struct PushNotificationChannel : PushNotificationChannelT<PushNotificationChannel, IWpnForegroundSink>
     {
         PushNotificationChannel(winrt::Windows::Networking::PushNotifications::PushNotificationChannel const& channel);
+
         winrt::Windows::Foundation::Uri Uri();
         winrt::Windows::Foundation::DateTime ExpirationTime();
         void Close();
 
         winrt::event_token PushReceived(winrt::Windows::Foundation::TypedEventHandler<Microsoft::Windows::PushNotifications::PushNotificationChannel, Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> handler);
         void PushReceived(winrt::event_token const& token) noexcept;
 
+        // IWpnForegroundSink
+        HRESULT __stdcall InvokeAll(ULONG length, byte* start) noexcept;
```

Comment:
This is byte* payload. Please lets not call it start. Also please reverse the order of the arguments. payload first and then length. Also missing SAL notation.